### ISSUE TITLE
Added dependency on Boost Chrono?

### ISSUE THREAD

**Issue by sehe** [libnonius/nonius#5]
Moving from rev 74b2de74c0f1a3f83d3f63 to ce74266b8878dbef42

breaks linking my nonius test program with link errors:

```
/tmp/direct-simple-8546e9.o: In function `nonius::timing<boost::chrono::steady_clock::duration, nonius::detail::complete_type<std::result_of<nonius::detail::repeater<nonius::now<boost::chrono::steady_clock> > (int)>::type>::type> nonius::detail::run_for_at_least<boost::chrono::steady_clock, nonius::detail::repeater<nonius::now<boost::chrono::steady_clock> > >(boost::chrono::steady_clock::duration, int, nonius::detail::repeater<nonius::now<boost::chrono::steady_clock> >&&)':
direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x15): undefined reference to `boost::chrono::steady_clock::now()'
direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x1d): undefined reference to `boost::chrono::steady_clock::now()'
direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x41): undefined reference to `boost::chrono::steady_clock::now()'
direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x51): undefined reference to `boost::chrono::steady_clock::now()'
direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x5b): undefined reference to `boost::chrono::steady_clock::now()'
/tmp/direct-simple-8546e9.o:direct-simple.cpp:(.text._ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_[_ZN6nonius6detail16run_for_at_leastIN5boost6chrono12steady_clockENS0_8repeaterINS_3nowIS4_EEEEEENS_6timingINT_8durationENS0_13complete_typeINSt9result_ofIFT0_iEE4typeEE4typeEEESB_iOSE_]+0x6a): more undefined references to `boost::chrono::steady_clock::now()' follow
```

Is this intentional? I hoped it could stay fully header-only.

Clang 3.5
Boost 1_57

**Comment by sehe** [comment]
PS. reproducers are in this gist: https://gist.github.com/sehe/54caed4b4cd7c9352174 (`make nonius` to build)

**Comment by rmartinho** [comment]
No, it's not intentional.

**Comment by rmartinho** [comment]
Fixed in 9567cfc8acc82f77b947f5b6641056fdfe6db78e
